icegathererstate.go

```diff
@@ -1,7 +1,11 @@
 package webrtc
 
+import (
+	"sync/atomic"
+)
+
 // ICEGathererState represents the current state of the ICE gatherer.
-type ICEGathererState byte
```

Comment:
Possible to not break public API? I would just make state in `icegatherer` atomic

Reply:
I also thought so, but using `atomic.Value` too much makes it more type unsafe system.
As it is typed and has enumerated const, I think assuming `ICEGathererState` as specific type in user code is a kind of unsupported thing. (If we should support this, we can't add any private member to the public struct. It also changes type.)

Reply:
That's a good point! Ok sounds good to me, approving :)

Sorry for so much churn on the PR, really appreciate everything you are doing.